src/main/java/duke/tasklist/DukeList.java

```diff
@@ -0,0 +1,131 @@
+package duke.tasklist;
+
+import duke.storage.Storage;
+import duke.task.Deadlines;
+import duke.task.Events;
+import duke.task.Task;
+
+import java.time.LocalDate;
+import java.util.List;
+
+public class DukeList {
+
+    private List<Task> a;
+    private Storage storage;
+
+    public DukeList(Storage s) {
+        this.a = s.load();
+        this.storage = s;
+    }
+
+    /**
+     * Adds a task to the list, prints add message to the console.
+     * @param t Task to be added
+     */
+    public void add(Task t) {
+        a.add(t);
+        storage.store(a);
+        System.out.println("\nDuke: Got it. I've added this task:\n      "
+                + t.show() + "\n      Now you have "
+                + this.a.size() + " tasks in the list.\n");
```

Comment:
Perhaps delegate the process of outputting messages to the UI class (as this could violate OOP principle)? I've noticed this issue in the other methods of this class as well.

Reply:
Hi, really appreciate all the comments that you've left, I think they were really helpful for me to identify the general problems with my code design and how I should improve. Will probably take me some time to fix the code design issues but thanks a lot!

Reply:
No problem. Glad it could help 👍